src/script/common/c_packer.cpp

```diff
@@ -358,22 +410,28 @@ static VectorRef<PackedInstr> pack_inner(lua_State *L, int idx, int vidx, Packed
 		else
 			rtable->uidata2++; // nrec
 
-		// check if we can use a shortcut
+		// set_into is a shortcut that allows a pushed value
+		// to be directly set into a table without separately pushing
+		// the key and using SETTABLE.
+		// only works in certain circumstances, hence the check:
 		if (can_set_into(ktype, vtype) && suitable_key(L, -2)) {
 			// push only the value
 			auto rval = pack_inner(L, absidx(L, -1), vidx, pv, seen);
+			vidx++;
```

Comment:
I don't get this change.
vidx is only used once between vidx++ and vidx--, and that one usage now has to use vidx - 1.
Unless I am missing something the old code was easier to read.

Reply:
It's functionally not different but more correct since after the `pack_inner` call space is taken up (`vidx++`) that is only freed after we make sure to pop it.
The old code smartly skipped this because the result is ±0 but I'm not convinced that's better for someone following along the code trying to understand it.